include/swift/AST/LayoutConstraint.h

```diff
@@ -108,6 +114,26 @@ class LayoutConstraintInfo : public llvm::FoldingSetNode {
     return isNativeRefCountedObject(Kind);
   }
 
+  bool isClass() const {
+    return isClass(Kind);
+  }
+
+  bool isNativeClass() const {
+    return isNativeClass(Kind);
+  }
+
+  bool isAnyClass() const {
+    return isAnyClass(Kind);
+  }
+
+  bool isRefCounted() const {
```

Comment:
These behavior the way I'd expect: `isRefCounted()` is `true` for any layout kind that ensures reference-counting.